### PR TITLE
Add extract_keyed_scalar_sum UDF

### PR DESCRIPTION
**Pull request by curtismorales**
Pulled this out of #3513 because that PR can't successfully build without this function existing. I plan to rerun bqetl_artifact_deployment after deploying this change, then #3513 

Checklist for reviewer:

- [ ] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title).
- [ ] If the PR comes from a fork, trigger integration CI tests by running the [Push to upstream workflow](https://github.com/mozilla/bigquery-etl/actions/workflows/push-to-upstream.yml) and provide the `<username>:<branch>` of the fork as parameter. The parameter will also show up
in the logs of the `manual-trigger-required-for-fork` CI task together with more detailed instructions.
- [ ] If adding a new field to a query, ensure that the schema and dependent downstream schemas have been updated.
- [ ] When adding a new derived dataset, ensure that data is not available already (fully or partially) and recommend extending an existing dataset in favor of creating new ones. Data can be available in the [bigquery-etl repository](https://github.com/mozilla/bigquery-etl), [looker-hub](https://github.com/mozilla/looker-hub) or in [looker-spoke-default](https://github.com/mozilla/looker-spoke-default/tree/e1315853507fc1ac6e78d252d53dc8df5f5f322b).

For modifications to schemas in restricted namespaces (see [`CODEOWNERS`](https://github.com/mozilla/bigquery-etl/blob/main/CODEOWNERS)):
- [ ] Follow the [change control procedure](https://docs.google.com/document/d/1TTJi4ht7NuzX6BPG_KTr6omaZg70cEpxe9xlpfnHj9k/edit#heading=h.ttegrcfy18ck)

### CODE REVIEW

File: sql/mozfun/map/extract_keyed_scalar_sum/udf.sql [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+)
+RETURNS INT64 AS (
+  (SELECT SUM(value) FROM UNNEST(keyed_scalar))
+);
+
+SELECT
+  assert.true(
+    map.extract_keyed_scalar_sum(
+      [STRUCT("a" AS key, 1 AS value), STRUCT("b" AS key, 2 AS value)]
+    ) = 3
```
Comment (fbertsch) [inline sql/mozfun/map/extract_keyed_scalar_sum/udf.sql:12]:
```suggestion
  assert.equals(
    3, map.extract_keyed_scalar_sum(
      [STRUCT("a" AS key, 1 AS value), STRUCT("b" AS key, 2 AS value)]
    )
```

File: sql/mozfun/map/extract_keyed_scalar_sum/udf.sql [hunk 2]
```diff
@@ -0,0 +1,13 @@
+CREATE OR REPLACE FUNCTION extract_keyed_scalar_sum(
+  keyed_scalar ARRAY<STRUCT<key STRING, value INT64>>
+)
+RETURNS INT64 AS (
+  (SELECT SUM(value) FROM UNNEST(keyed_scalar))
+);
+
+SELECT
```
Comment (fbertsch) [inline sql/mozfun/map/extract_keyed_scalar_sum/udf.sql:8]: Tests are cheap! Consider adding some edge cases, e.g. a `NULL` value, `NULL` input, etc.

**Review by fbertsch — APPROVED** [review]
r+wc
